lib/collection.js

```diff
@@ -944,9 +944,26 @@ define.classMethod('insert', { callback: true, promise: true });
  */
 Collection.prototype.updateOne = function(filter, update, options, callback) {
   var self = this;
+
   if (typeof options == 'function') (callback = options), (options = {});
   options = shallowClone(options);
 
+  var keys = Object.keys(update);
+
+  // same errors as the server would give for update doc lacking atomic operators
+  var err;
+
+  if (keys.length == 0) {
+    err = 'The update operation document must contain at least one atomic operator.';
+  } else if (keys[0][0] != '$') {
```

Comment:
I don't know how you want to force that at least one atomic operator has to exist,
but I guess somehow you have to check all the update operators,
not only the first one that ```Object.keys``` returns.

You should instead use something like:
```
    var at_least_one = keys.some(function(keyname){ return keyname[0] === '$'; })
```
or 

```
    var everyone_does = keys.every(function(keyname){ return keyname[0] === '$'; })
```


depending on if you want all of them to be atomic or at least one of them.

Reply:
@Loksly not a bad idea, and one we had considered, however this approach is a) potentially more performant, and b) spec compliant. Per the CRUD spec:
```
The update family of operations require that the update document parameter MUST have only atomic modifiers. In practice, this means that introspection needs to happen on that document to enforce this. However, it is enough to only check the first element in the document. If it begins with a $ sign and the rest of the document's elements do not, the server will throw an error. Note that it is required that an update document have at least one atomic modifier.
```

This is also in line with the shells behavior.